static/app/views/starfish/components/samplesTable/useSlowMedianFastSamplesQuery.tsx

```diff
@@ -0,0 +1,163 @@
+import EventView from 'sentry/utils/discover/eventView';
+import {QueryFieldValue} from 'sentry/utils/discover/fields';
+import {
+  DiscoverQueryProps,
+  useGenericDiscoverQuery,
+} from 'sentry/utils/discover/genericDiscoverQuery';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const LIMIT_PER_POPULATION = 5;
+
+/**
+ * This hook will fetch transaction events from 3 different types of populations and combine them in one set, then return them:
+ *
+ * - Slowest Events
+ * - Median / Baseline Events
+ * - Fastest Events
+ *
+ * It assumes that you are passing an eventView object with a query scoped to a specific transaction
+ *
+ * @param eventView An eventView containing query information, such as the transaction and other filters
+ */
+export default function useSlowMedianFastSamplesQuery(eventView: EventView) {
+  const location = useLocation();
+  const organization = useOrganization();
+
+  const commonColumns: QueryFieldValue[] = [
+    {
+      field: 'transaction.duration',
+      kind: 'field',
+    },
+    {
+      field: 'profile_id',
+      kind: 'field',
+    },
+    {
+      field: 'timestamp',
+      kind: 'field',
+    },
+  ];
+
+  const eventViewAggregates = eventView.clone().withColumns([
+    {kind: 'function', function: ['p50', 'transaction.duration', undefined, undefined]},
+    {kind: 'function', function: ['p95', 'transaction.duration', undefined, undefined]},
+  ]);
+
+  const {isLoading: isLoadingAgg, data: aggregatesData} = useGenericDiscoverQuery<
+    any,
+    DiscoverQueryProps
+  >({
+    route: 'events',
+    eventView: eventViewAggregates,
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+  });
+
+  const slowestSamplesEventView = eventView
+    .clone()
+    .withColumns(commonColumns)
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'desc',
+      },
+    ]);
+
+  slowestSamplesEventView.additionalConditions = new MutableSearch(
+    `transaction.duration:>${
+      aggregatesData?.data?.[0]?.['p95(transaction.duration)'] ?? 0
+    }`
+  );
+
+  const {isLoading: isLoadingSlowest, data: slowestSamplesData} = useGenericDiscoverQuery<
+    any,
+    DiscoverQueryProps
+  >({
+    route: 'events',
+    eventView: slowestSamplesEventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...slowestSamplesEventView.getEventsAPIPayload(location),
+    }),
```

Comment:
This shouldn't be needed because `GenericDiscoverQuery` will call this for you
https://github.com/getsentry/sentry/blob/d7e0b08f848bf29f082721b511f0c2da1ded54b2/static/app/utils/discover/genericDiscoverQuery.tsx#L364

Reply:
Ah good catch, this was copypasta'd from the old code. It works fine without it so I'll remove it